plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/FileSystemExchangeStorage.java

```diff
@@ -0,0 +1,52 @@
+package io.trino.plugin.exchange;
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import io.airlift.slice.SliceInput;
+
+import javax.crypto.SecretKey;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public interface FileSystemExchangeStorage
+        extends AutoCloseable
+{
+    void initialize(URI baseDirectory);
+
+    void createDirectories(URI dir) throws IOException;
+
+    SliceInput getSliceInput(URI file, Optional<SecretKey> secretKey) throws IOException;
```

Comment:
Similarly to https://github.com/trinodb/trino/pull/10376/commits/3558ba211af999233ed1a9a3517b418fa6b87be0#r774711784 FileSystemExchangeStorage should be able to tell that reading is blocked right now and this information should be passed on to `FileSystemExchangeSource.isBlocked()`

Mybe a followup.

Reply:
Yes this is on the list